.github/workflows/build.yml

```diff
@@ -0,0 +1,23 @@
+name: "Build"
+on: push
+jobs:
+  test_ros:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - run: |
+        npm ci
+        npm run build
+    - run: |
+        git checkout --detach
+        git config user.email "ros-tooling@users.noreply.github.com"
+        git config user.name "ROS Tooling [bot]"
+        git add --force dist 
+        git commit --no-verify -m "$commitmsg" --only dist
+        git push origin --no-verify HEAD:refs/tags/build/${{github.sha}}
```

Comment:
I was thinking we would push to a `release` branch. It would need a few additional files (`action.yml`, `README`, `LICENSE`) - I think the best way would be to have it be a pure-augmentation of the `master` branch. So maybe it would look something like

```
# dist is never checked in on master
npm run build
git checkout release
git rebase master
git add dist/index.js
git commit -m "Build artifacts for ${{ github.sha }}"
git push origin release
```

We would then only create tags when we actually want to release out a new version of the action, which could be manual or on an otherwise automated schedule. If `release` naming is confusing then maybe we call the branch `staging`?

Reply:
It won't do to have a `release` branch.

1. That doesn't solve the problem for building PRs.
2. Workflow runs are not guaranteed to be sequential

We'd basically have to have a shadow `build/xxx` branch for every `xxx` branch and ensure that they run in order even when simultaneous commits are run.